spikeinterface/sortingcomponents/waveforms/temporal_pca.py

```diff
@@ -0,0 +1,231 @@
+            exception_string = (
+                f"model_path folder is not a folder or does not exist. \n"
+                f"A model can be trained by using{self.__class__.__name__}.fit(...)"
+            )
+            raise TypeError(exception_string)
+
+        # Load the model and the time interval dict from the model_folder
+        model_path = Path(model_folder_path) / "pca_model.pkl"
+        with open(model_path, "rb") as f:
+            self.pca_model = pickle.load(f)
+        model_time_interval_path = Path(model_folder_path) / "time_interval.json"
+        with open(model_time_interval_path, "rb") as f:
+            self.waveform_time_interval_dict = json.load(f)
+
+        # Find waveform extractor in the parents
+        try: 
+            waveform_extractor = next(parent for parent in self.parents if isinstance(parent, WaveformExtractorNode))
+        except (StopIteration, TypeError):
+            exception_string = f"TemporalPCA should have a {WaveformExtractorNode.__name__} in its parents"
+            raise TypeError(exception_string)
+
+        self.assert_model_and_waveform_temporal_match(waveform_extractor)
+    
+    def assert_model_and_waveform_temporal_match(self, waveform_extractor: WaveformExtractorNode):
+        """
+        Asserts that the model and the waveform extractor have the same temporal parameters
+        """        
+        # Extract the first waveform extractor in the parents
+        waveforms_ms_before = waveform_extractor.ms_before
+        waveforms_ms_after = waveform_extractor.ms_after
+        waveforms_sampling_frequency = waveform_extractor.recording.get_sampling_frequency()
+
+        model_ms_before = self.waveform_time_interval_dict["ms_before"]
+        model_ms_after= self.waveform_time_interval_dict["ms_after"]
+        model_sampling_frequency = self.waveform_time_interval_dict["sampling_frequency"]
+        
+        ms_before_mismatch = waveforms_ms_before != model_ms_before
+        ms_after_missmatch = waveforms_ms_after != model_ms_after
+        sampling_frequency_mismatch = waveforms_sampling_frequency != model_sampling_frequency
+        if ms_before_mismatch or ms_after_missmatch or sampling_frequency_mismatch:
+            exception_string = (
+                "PCA model and waveforms mismatch \n"
+                f"{model_ms_before=} and {waveforms_ms_after=} \n"
+                f"{model_ms_after=} and {waveforms_ms_after=} \n"
+                f"{model_sampling_frequency=} and {waveforms_sampling_frequency=} \n"
+            )
+            raise ValueError(exception_string)
+    
+    @staticmethod
+    def fit(
+        recording: BaseRecording,
+        n_components: int,
+        model_folder_path: str,
+        detect_peaks_params: dict,
+        peak_selection_params: dict,
+        job_kwargs: dict = None,
+        ms_before: float = 1.0,
+        ms_after: float = 1.0,
+        whiten: bool = True,
+        local_radius_um: float = None,
+    ) -> IncrementalPCA:
```

Comment:
Could we also add the predetected peaks in params  to skip the detect and slelect ?